### PR TITLE
Update Molecule config for local use

### PR DESCRIPTION
**Pull request by lewisdenny**
- Change container name to instance to match what runs in CI so scenarios can be written to run locally and in CI without tweaks
- Remove unused TRIPLEO_ANSIBLE overrides
- Remove unused YAML anchors

### CODE REVIEW

**Comment by lewisdenny** [comment]
recheck

**Comment by lewisdenny** [comment]
recheck

**Comment by lewisdenny** [comment]
recheck

**Comment by lewisdenny** [comment]
recheck
added hold

**Comment by lewisdenny** [comment]
recheck

**Comment by lewisdenny** [comment]
@openstack-k8s-operators/cifmw-squad Hey folks, requesting an override for this patch -_-

File: .config/molecule/config_podman.yml [hunk 1]
```diff
@@ -5,42 +5,30 @@ driver:
...
   env:
     ANSIBLE_STDOUT_CALLBACK: yaml
-    ANSIBLE_ROLES_PATH: "${ANSIBLE_ROLES_PATH}:${HOME}/zuul-jobs/roles"
-    ANSIBLE_LIBRARY: "${ANSIBLE_LIBRARY:-/usr/share/ansible/plugins/modules}"
-    ANSIBLE_FILTER_PLUGINS: "${ANSIBLE_FILTER_PLUGINS:-/usr/share/ansible/plugins/filter}"
   config_options:
     defaults:
       remote_tmp: /tmp
-
-scenario:
```
Comment (frenzyfriday) [inline .config/molecule/config_podman.yml:42]: lgtm, but I am trying to get some context:
what was this scenario being used for?

Comment (lewisdenny) [reply .config/molecule/config_podman.yml:42]: So how the Molecule test works is it merges the configs together, each test under roles/<rolename>/molecule/* has it's own partial config. The idea being that `config_podman.yml` holds the common config for all the tests and then the test overrides/sets what is needed for the role. For example setting hostvars. 

Scenarios is not something that should be defined as a default but left to each test to set themselves :)

**Pull request merged**